File: nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/presenter/impl/base/model/FiscalYear_Ds.java
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.presenter.impl.base.model;

import java.util.Date;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.Param;
import net.nan21.dnet.core.api.annotation.RefLookup;
import net.nan21.dnet.core.api.annotation.RefLookups;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.AbstractTypeDs;
import net.nan21.dnet.module.md.domain.impl.base.Calendar;
import net.nan21.dnet.module.md.domain.impl.base.FiscalYear;

@Ds(entity = FiscalYear.class, sort = {@SortField(field = FiscalYear_Ds.f_startDate, desc = true)})
@RefLookups({@RefLookup(refId = FiscalYear_Ds.f_calendarId, namedQuery = Calendar.NQ_FIND_BY_NAME, params = {@Param(name = "name", field = FiscalYear_Ds.f_calendar)})})
public class FiscalYear_Ds extends AbstractTypeDs<FiscalYear> {

	public static final String f_startDate = "startDate";
	public static final String f_endDate = "endDate";
	public static final String f_calendarId = "calendarId";
	public static final String f_calendar = "calendar";

	@DsField
	private Date startDate;

	@DsField
	private Date endDate;

	@DsField(join = "left", path = "calendar.id")
	private String calendarId;

	@DsField(join = "left", path = "calendar.name")
	private String calendar;

	public FiscalYear_Ds() {
		super();
	}

	public FiscalYear_Ds(FiscalYear e) {
		super(e);
	}

	public Date getStartDate() {
		return this.startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public Date getEndDate() {
		return this.endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public String getCalendarId() {
		return this.calendarId;
	}

	public void setCalendarId(String calendarId) {
		this.calendarId = calendarId;
	}

	public String getCalendar() {
		return this.calendar;
	}

	public void setCalendar(String calendar) {
		this.calendar = calendar;
	}
}

File: nan21.dnet.module.md.ui.extjs/src/main/resources/webapp/js/ds/FiscalYear_Ds.js
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
	
Ext.define(Dnet.ns.md + "FiscalYear_Ds", {
	extend: 'Ext.data.Model',
	
	validations: [
		{field: "calendar", type: 'presence'},
		{field: "name", type: 'presence'},
		{field: "startDate", type: 'presence'},
		{field: "endDate", type: 'presence'}
	],
	
	fields: [
		{name:"startDate", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"endDate", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"calendarId", type:"string"},
		{name:"calendar", type:"string"},
		{name:"id", type:"string"},
		{name:"name", type:"string"},
		{name:"description", type:"string"},
		{name:"notes", type:"string"},
		{name:"active", type:"boolean"},
		{name:"clientId", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"refid", type:"string"},
		{name:"entityAlias", type:"string"},
		{name:"entityFqn", type:"string"}
	]
});
